File: USACO/namenum.cpp
/*
ID: 2010pes1
TASK: namenum
LANG: C++
*/

#include "bits/stdc++.h"

using namespace std;

int main(){
    string myline = "";
    fstream myfile;

    myfile.open("dict.txt", ios::in);
    if(myfile.is_open()){
       string tp;
       while(getline(myfile, tp)){
           myline += tp + "\n";
       }
       myfile.close(); //close the file object
    }
    
    map<int,string> m;
    m[2] = "ABC";
    m[3] = "DEF";
    m[4] = "GHI";
    m[5] = "JKL";
    m[6] = "MNO";
    m[7] = "PRS";
    m[8] = "TUV";
    m[9] = "WXY";

    /*
          2: A,B,C     5: J,K,L    8: T,U,V
          3: D,E,F     6: M,N,O    9: W,X,Y
          4: G,H,I     7: P,R,S
    */

    freopen("namenum.in", "r", stdin);
    freopen("namenum.out", "w", stdout);
    string s;
    scanf("%s",&s);
    int n = s.length();
    for(int i = 0; i<n;i++){
        if(m[atoi(s[i])]){
        
        }
    }
    return 0;
}

File: USACO/namenum.py
'''
ID: 2010pes1
TASK: namenum
LANG: PYTHON3
'''

def solve(lst, dial, rdial, numstr):
    mklst = []
    preplst = [i for i in lst if (len(i) == len(numstr))] 
    for word in preplst:
        recstr = ''
        for ch in word:
            if ch in rdial:
                recstr += rdial[ch]
        if numstr == recstr:
            mklst.append(word)
    return mklst

def main():
    dial = {}
    f = open("dict.txt","r")

    lst = f.read().strip().split('\n')
    #print(lst) 
    f.close()
    f = open("namenum.in", "r")
    numstr = f.read().strip()
    f.close()

    dial['2'] = "ABC"
    dial['3'] = "DEF"
    dial['4'] = "GHI"
    dial['5'] = "JKL"
    dial['6'] = "MNO"
    dial['7'] = "PRS"
    dial['8'] = "TUV"
    dial['9'] = "WXY"

    rdial = {}
    for i in dial:
        for j in range(3):
            rdial[dial[i][j]] = i
    
    f = open("namenum.out", "w")
    ans = solve(lst,dial,rdial,numstr)
    if len(ans) == 0:
        f.write(str("NONE")+'\n')
    else:
        for i in ans:
            f.write(str(i)+'\n')
    f.close()

if __name__ == "__main__":
    main()
